whylog_vim/proxy/__init__.py

```diff
@@ -0,0 +1,64 @@
+from whylog import whylog_factory
+from whylog_vim.proxy.teacher import TeacherProxy
+from whylog_vim.proxy.log_reader import LogReaderProxy
+from whylog_vim.consts import EditorStates as States, ActionTypes
+from whylog_vim.exceptions import UnknownAction
+
+
+class WhylogProxy(object):
+    def __init__(self, editor):
+        self.editor = editor
+        log_reader, teacher_generator = whylog_factory()
+        self.teacher_generator = teacher_generator
+        self.log_reader = LogReaderProxy(editor, log_reader)
+        self.new_teacher()
```

Comment:
It would be more clear if you write in `__init__` that there exists teacher field. Otherwise, I see in line 26 something about self.teacher and must search it in code. Wouldn't it be better to write in __init__ something like: `self.teacher = ...` and `self._state = States.EDITOR_NORMAL` as you have in new_teacher method?